### PR TITLE
Live location sharing: don't group beacon info with room creation summary

### PR DESCRIPTION
**Pull request by kerryarchibald**
As state events, when a beacon is created in an empty room it is grouped in the creation summary.
Beacons are not part of room setup and should appear in the timeline, don't group them in creation summary.

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Live location sharing: don't group beacon info with room creation summary ([\#8468](https://github.com/matrix-org/matrix-react-sdk/pull/8468)). Contributed by @kerryarchibald.

### CODE REVIEW

**Review by t3chguy — COMMENTED** [review]
The code looks to do what the PR says, but there's no explanation anywhere as to *why*

**Comment by kerryarchibald** [comment]
@t3chguy added some context.

File: src/components/structures/MessagePanel.tsx [hunk 1]
```diff
@@ -1098,9 +1099,15 @@ class CreationGrouper extends BaseGrouper {
             && (ev.getStateKey() !== createEvent.getSender() || ev.getContent()["membership"] !== "join")) {
             return false;
         }
+        // beacons are not part of room creation configuration
+        // should be shown in timeline
+        if (M_BEACON_INFO.matches(ev.getType())) {
+            return false;
+        }
```
Comment (t3chguy) [inline src/components/structures/MessagePanel.tsx:1106]: The comment on lines L1081-1083 needs updating too

**Review by t3chguy — CHANGES_REQUESTED** [review]

**Review by t3chguy — APPROVED** [review]
Thanks!

**Pull request merged**